libraries/chain/include/steemit/chain/database.hpp

```diff
@@ -422,7 +422,7 @@ namespace steemit { namespace chain {
       private:
          optional< chainbase::database::session > _pending_tx_session;
 
-         void apply_block( const signed_block& next_block, uint32_t skip = skip_nothing );
+         void apply_block( const signed_block& next_block, uint32_t skip = skip_nothing, uint32_t current_block_num = 0 );
```

Comment:
This is bad. Many places that currently call apply_block would now have broken error messages. Future code may begin to rely on that value instead of grabbing the block num from the block itself and break because of it.